### ISSUE TITLE
Code generation bug with generics

### ISSUE THREAD

**Issue by peter-perot** [dotnet/orleans#5340]
The following interface and implementation cause a compile-time error with Orleans 1.5.6:

``` csharp
public interface IBugWithGenerics : IGrainWithIntegerKey
{
  Task SomeMethod<T>(string argument);
  Task SomeMethod<T>(int argument);
}

public class BugwithGenerics : Grain, IBugWithGenerics
{
  public Task SomeMethod<T>(string argument)
  {
    throw new NotImplementedException();
  }

  public Task SomeMethod<T>(int argument)
  {
    throw new NotImplementedException();
  }
}
```

The compile-time error is here:

``` csharp
[global::System.CodeDom.Compiler.GeneratedCodeAttribute(@"Orleans-CodeGenerator", @"1.5.6.0"), global::Orleans.CodeGeneration.MethodInvokerAttribute(typeof(global::Implico.WebTemplate.Services.Tests.Faking.DataAccessLayer.MockEvidenceServices.IBugWithGenerics), -2136338972), global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute]
internal class OrleansCodeGenBugWithGenericsMethodInvoker : global::Orleans.CodeGeneration.IGrainMethodInvoker
{
  private static readonly global::Orleans.CodeGeneration.GenericMethodInvoker SomeMethodT = new global::Orleans.CodeGeneration.GenericMethodInvoker(typeof(global::Implico.WebTemplate.Services.Tests.Faking.DataAccessLayer.MockEvidenceServices.IBugWithGenerics), @"SomeMethod", 1);
  private static readonly global::Orleans.CodeGeneration.GenericMethodInvoker SomeMethodT = new global::Orleans.CodeGeneration.GenericMethodInvoker(typeof(global::Implico.WebTemplate.Services.Tests.Faking.DataAccessLayer.MockEvidenceServices.IBugWithGenerics), @"SomeMethod", 1);
  public global::System.Threading.Tasks.Task<global::System.Object> @Invoke(global::Orleans.Runtime.IAddressable @grain, global::Orleans.CodeGeneration.InvokeMethodRequest @request)
  {
    global::System.Int32 interfaceId = @request.@InterfaceId;
    global::System.Int32 methodId = @request.@MethodId;
    global::System.Object[] arguments = @request.@Arguments;
    if (@grain == null)
      throw new global::System.ArgumentNullException(@"grain");
    switch (interfaceId)
    {
      case -2136338972:
        switch (methodId)
        {
          case 1206537793:
            return SomeMethodT.@Invoke(@grain, arguments);
          case 1029784630:
            return SomeMethodT.@Invoke(@grain, arguments);
          default:
            throw new global::System.NotImplementedException(@"interfaceId=" + -2136338972 + @",methodId=" + methodId);
        }

      default:
        throw new global::System.NotImplementedException(@"interfaceId=" + interfaceId);
    }
  }

  public global::System.Int32 InterfaceId
  {
    get
    {
      return -2136338972;
    }
  }

  public global::System.UInt16 InterfaceVersion
  {
    get
    {
      return 1;
    }
  }
}
```

There are two generic method invokers named `SomeMethodT` generated which does not compile.

**Comment by ReubenBond** [comment]
Thanks for reporting, @peter-perot. That's certainly a bug. For now the workaround may be to change the type parameter name of one of the overloads

**Comment by ReubenBond** [comment]
Fixed by #5340 (fixing this was required for the tests added in that PR to work 😅)
